modules/spring-data/src/main/java/org/apache/ignite/springdata/repository/support/IgniteRepositoryImpl.java

```diff
@@ -101,20 +102,37 @@ public IgniteRepositoryImpl(IgniteCache<ID, T> cache) {
         };
     }
 
-    /** {@inheritDoc} */
-    @Override public Iterable<T> findAll(Iterable<ID> ids) {
+    /**
+     * @param ids Collection of IDs.
+     * @return Collection transformed to set.
+     */
+    private Set<ID> toSet(Iterable<ID> ids) {
         if (ids instanceof Set)
-            return cache.getAll((Set<ID>)ids).values();
+            return (Set<ID>)ids;
+
+        Iterator<ID> itr = ids.iterator();
+
+        if (!itr.hasNext())
+            return emptySet();
+
+        ID key = itr.next();
+
+        Set<ID> keys = key instanceof Comparable ? new TreeSet<>() : new HashSet<>();
```

Comment:
What happens if not all keys are comparable? I understand this is rare and undesirable case but it is possible. We should at least reflect it somewhere.

Reply:
@alamar This is not possible, because repository declared as Repository<V, K>
Where "K" - is a type for keys, so all keys will be of same type.